apps/application-system/api/src/app/modules/application/files/file.service.ts

```diff
@@ -71,14 +73,18 @@ export class FileService {
   ): Promise<SigningServiceResponse> {
     this.validateApplicationType(application.typeId)
     const { answers, externalData, id, state } = application as CRCApplication
-    const { nationalRegistry, parentNationalRegistry } = externalData
+    const { nationalRegistry } = externalData
     const isParentA = state === 'draft'
-
-    const parentBName = parentNationalRegistry.data.name
+    const applicant = nationalRegistry?.data
+    const selectedChildren = getSelectedChildrenFromExternalData(
+      applicant.children,
+      answers.selectChild,
+    )
+    const parentB = selectedChildren[0].otherParent
```

Comment:
Should we guard here against an empty array? 
Know it should not be possible with the current flow, but a thought.

Reply:
Well I don't think we need to guard aginst this. You shouldn't be able to get to this step without this being in place. You will be stopped earlier in the flow. I have also removed other unneccessary guards.